spyder_kernels/console/tests/test_console_kernel.py

```diff
@@ -811,20 +812,66 @@ def test_comprehensions_with_locals_in_pdb(kernel):
     pdb_obj = SpyderPdb()
     pdb_obj.curframe = inspect.currentframe()
     pdb_obj.curframe_locals = pdb_obj.curframe.f_locals
-    kernel._pdb_obj = pdb_obj
+    kernel.shell.pdb_session = pdb_obj
 
     # Create a local variable.
-    kernel._pdb_obj.default('zz = 10')
+    kernel.shell.pdb_session.default('zz = 10')
     assert kernel.get_value('zz') == 10
 
     # Run a list comprehension with this variable.
-    kernel._pdb_obj.default("compr = [zz * i for i in [1, 2, 3]]")
+    kernel.shell.pdb_session.default("compr = [zz * i for i in [1, 2, 3]]")
     assert kernel.get_value('compr') == [10, 20, 30]
 
     # Check that the variable is not reported as being part of globals.
-    kernel._pdb_obj.default("in_globals = 'zz' in globals()")
+    kernel.shell.pdb_session.default("in_globals = 'zz' in globals()")
     assert kernel.get_value('in_globals') == False
 
+    pdb_obj.curframe = None
+    pdb_obj.curframe_locals = None
+
+
+def test_namespaces_in_pdb(kernel):
+    """
+    Test namespaces in pdb
+    """
+    # Define get_ipython for timeit
+    get_ipython = lambda: kernel.shell
+    kernel.shell.user_ns["test"] = 0
+    pdb_obj = SpyderPdb()
+    pdb_obj.curframe = inspect.currentframe()
+    pdb_obj.curframe_locals = pdb_obj.curframe.f_locals
+    kernel.shell.pdb_session = pdb_obj
+    # Check adding something to globals works
```

Comment:
```suggestion

    # Check adding something to globals works
```

Leave blank for clarity.